test/widgets/content_test.dart

```diff
@@ -290,6 +297,79 @@ void main() {
 
   testContentSmoke(ContentExample.mathBlock);
 
+  testContentSmoke(ContentExample.strong);
+
+  testContentSmoke(ContentExample.emphasis);
+
+  group('InlineCodeNode', () {
+    testContentSmoke(ContentExample.inlineCode);
+
+    testWidgets('maintains font-size ratio with surrounding text', (tester) async {
+      /// Simulate a nested span's style by merging all ancestor-span styles,
+      /// starting from the root.
+      ///
+      /// This isn't how the style actually gets applied in the code under test,
+      /// but it should hopefully simulate it closely enough.
+      TextStyle? mergeStylesOuterToInner(InlineSpan outerSpan, InlineSpan innerSpan) {
+        final styles = <TextStyle?>[];
+        bool recurse(InlineSpan span) {
+          styles.add(span.style);
+          if (span == innerSpan) {
+            return false;
+          }
+          final notInSubtree = span.visitDirectChildren(recurse);
+          if (notInSubtree) {
+            styles.removeLast();
+          }
+          return notInSubtree;
+        }
+        recurse(outerSpan);
+
+        return styles.reduce((value, element) => switch ((value, element)) {
+          (TextStyle(), TextStyle()) => value!.merge(element!),
```

Comment:
After the comment above about the algorithm, I notice that these tests pass just the same if I reverse the order these get merged in. It's fairly subtle to get that ordering the right way around, so it'd be good to have a check to prevent it accidentally breaking. (When merged in the wrong order, all the fonts are the same — so the ratio check always passes, without successfully testing what we want to test.)

Probably the simplest way to do that is to check that the font size is bigger in the header than the paragraph, and bigger in the plain text than the inline code — say by at least 10% in each case. That's definitely a solid expectation for header vs. paragraph, and mostly solid for plain vs. code; and if someday we change the code style so the latter no longer holds, we can figure out a different check.